controllers/cart/getCart.js

```diff
@@ -0,0 +1,34 @@
+const Cart = require('../../models/cart')
+const Store = require('../../models/store')
+const Product = require('../../models/product')
+
+async function getCart(req, res) {
+  try {
+    const storeId = req.params.store_id
+    const store = await Store.findById({ _id: storeId })
+    if (!store) {
+      return res.status(404).json({ message: 'No store found' })
+    }
+
+    const populatedCart = await Cart.aggregate([
+      {
+        $lookup: {
+          from: 'products',
+          localField: 'contents',
+          foreignField: '_id',
+          as: 'productObjects'
+        }
+      }
+    ])
+
+    const storeCart = populatedCart.filter(
+      item => String(item.storeId) === String(storeId)
+    )
+
+    res.status(200).json(storeCart)
```

Comment:
Will this return something like this
```
   {
  contents: [
    { id: 6789087343, price: 400, description: "Fine product", name: "Rice" },
    { id: 6789087343, price: 400, description: "Fine product", name: "Rice" }
  ],
  checkedOut: false,
  _id: "5e1df5f957aff215deecc4b7",
  agreedPrice: 45,
  total: 540,
  storeId: "5e1def4257aff215deecc4ac",
  checkoutDate: "2020-01-14T17:10:17.173Z",
  __v: 0
};
```

Reply:
yeah. this is a sample output. the contents are expanded into a property called `productObjects`
```
{
    "_id": "5e1e192390f5d236c6107b04",
    "contents": [
      "5e1defb857aff215deecc4ad",
      "5e1df13b57aff215deecc4ae",
      "5e1df14457aff215deecc4af"
    ],
    "checkedOut": false,
    "agreedPrice": 45,
    "total": 540,
    "storeId": "5e1def4257aff215deecc4ac",
    "checkoutDate": "2020-01-14T19:40:19.557Z",
    "__v": 0,
    "productObjects": [
      {
        "_id": "5e1defb857aff215deecc4ad",
        "images": [
          "mee.jpg",
          "us.jpg"
        ],
        "name": "Shoes",
        "description": "A very nice ",
        "price": 500,
        "stock": 10,
        "storeId": "5e1def4257aff215deecc4ac",
        "__v": 0
      },
      {
        "_id": "5e1df13b57aff215deecc4ae",
        "images": [
          "mee.jpg",
          "us.jpg"
        ],
        "name": "Shoes1",
        "description": "A very nice ",
        "price": 500,
        "stock": 10,
        "storeId": "5e1def4257aff215deecc4ac",
        "__v": 0
      },
      {
        "_id": "5e1df14457aff215deecc4af",
        "images": [
          "mee.jpg",
          "us.jpg"
        ],
        "name": "Shoes2",
        "description": "A very nice ",
        "price": 500,
        "stock": 10,
        "storeId": "5e1def4257aff215deecc4ac",
        "__v": 0
      }
    ]
  }
```

Reply:
I feel the `productObjects` should replace `contents`